api/src/main/java/org/openmrs/api/db/hibernate/HibernateAdministrationDAO.java

```diff
@@ -203,18 +211,82 @@ public void setApplicationContext(ApplicationContext applicationContext) throws
 	 */
 	@Override
 	public void validate(Object object, Errors errors) throws DAOException {
+		
+		Class entityClass = object.getClass();
+
+		ClassMetadata metadata = sessionFactory.getClassMetadata(entityClass);
+		if (metadata != null) {
+			String[] propNames = metadata.getPropertyNames();
+			Object IdentifierTpye = metadata.getIdentifierType();
+			String identifierName = metadata.getIdentifierPropertyName();
+			if (IdentifierTpye instanceof StringType && identifierName != null) {
+				ValidateUtil.validateFieldLengths(errors, object.getClass(), identifierName);
+			}
+			for (int i = 0; i < propNames.length; i++) {
+				String propType = metadata.getPropertyType(propNames[i]).getName();
+				if (propType.equalsIgnoreCase("String")) {
+					Field fieldTerm = null;
+					try {
+						ValidateUtil.validateFieldLengths(errors, object.getClass(), propNames[i]);
```

Comment:
I starting to think you should get the value from the class metadata and you don't delegate to ValidateUtil so that you don't need the reflective code in case there are fields with no getters